src/client/app/modules/analytics/analytics.controller.js

```diff
@@ -0,0 +1,156 @@
+;(function() {
+
+  angular
+    .module('app.controllers.AnalyticsController', [])
+    .controller('AnalyticsController', AnalyticsController);
+  
+  /* @ngInject */
+  function AnalyticsController($scope, $stateParams, user, routes) {
+    var vm = this;
+
+  
+    // form info regarding this route
+    vm.formInfo = {
+      route: $stateParams.route,
+      methodKeys: [],
+      methods: {},
+      data: [],
+      persistence: false,
+      businessLogic: ''
+    };
+
+    vm.sortType     = 'id'; // set the default sort type
+    vm.sortReverse  = false;  // set the default sort order
+    vm.searchData   = '';     // set the default search/filter term
+  
+
+    vm.tableHead = vm.formInfo.data[0];
+    vm.prepareData = prepareData;
+    vm.analytics = {};
+
+    activate();
+
+
+    function activate() {
+      // getRoute();
+      getRoute().then(function() {
+        console.log('route received');
+        var data = prepareData();
+        renderChart(data);
+      });
+    }
+
+    function getRoute() {
+      return routes.getRoute(vm.formInfo.route)
+        .then(function(res) {
+          /**
+           * The endpoint DB model stores response, headers, status, etc., all
+           * on the `methods` property. Here we're separating out method keys from
+           * the body for the sake of ng-repeat in view.
+           */
+          vm.formInfo.methods = res.methods;
+          vm.formInfo.persistence = res.persistence;
+          vm.formInfo.data = res.data;
+          vm.formInfo.methodKeys = Object.keys(vm.formInfo.methods);
+          vm.formInfo.businessLogic = res.businessLogic;
+          vm.tableHead = vm.formInfo.data[0];
+          if(res.analytics) {
+            vm.analytics = res.analytics;
+          }
+          console.log('analytics data from db' , res.analytics);
+          return;
+        }).catch(function(err) {
+          console.error('error fetching route', vm.formInfo.route);
+          console.error('Message:', err);
+        });
+    }
+
+    function prepareData() {
+      var position = {
+        'GET': 0,
+        'POST': 1,
+        'PUT': 2,
+        'DELETE': 3
+      };
+
+      var defaultValues = {
```

Comment:
It's unclear (from reading your code only) what these default values mean. I would add some documentation here to explain. Same with `position` above.